From: Ben Hutchings <ben@decadent.org.uk>
Subject: iwlegacy: Use consistent condition for il_pm_ops
Date: Tue, 11 Jun 2013 04:37:26 +0100

Commit e82add550d9b 'net: wireless: iwlegacy: add CONFIG_PM_SLEEP to
suspend/resume functions' correctly changed the condition for defining
il_pm_ops and the functions it points to.  But it didn't change the
condition for use of il_pm_ops, so iwl3945 and iwl4965 fail to link if
only CONFIG_PM is enabled.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
--- a/drivers/net/wireless/iwlegacy/common.h
+++ b/drivers/net/wireless/iwlegacy/common.h
@@ -1832,16 +1832,16 @@ u32 il_usecs_to_beacons(struct il_priv *
 __le32 il_add_beacon_time(struct il_priv *il, u32 base, u32 addon,
 			  u32 beacon_interval);
 
-#ifdef CONFIG_PM
+#ifdef CONFIG_PM_SLEEP
 extern const struct dev_pm_ops il_pm_ops;
 
 #define IL_LEGACY_PM_OPS	(&il_pm_ops)
 
-#else /* !CONFIG_PM */
+#else /* !CONFIG_PM_SLEEP */
 
 #define IL_LEGACY_PM_OPS	NULL
 
-#endif /* !CONFIG_PM */
+#endif /* !CONFIG_PM_SLEEP */
 
 /*****************************************************
 *  Error Handling Debugging
